org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/matrices/GithubApiCaller.java

```diff
@@ -0,0 +1,132 @@
+     * This is used for calling the github search REST API.
+     *
+     * @param commitHash        commit hash to be searched
+     * @param githubAccessToken Github access token for accessing github API
+     * @return String representation of the json response
+     * @throws CodeQualityMatricesException
+     */
+    public String callSearchCommitApi(String commitHash, String githubAccessToken) throws CodeQualityMatricesException {
+        String url = "https://api.github.com/search/commits?q=hash%3A" + commitHash;
+        try {
+            httpGet = new HttpGet(url);
+            httpGet.addHeader("Authorization", "Bearer " + githubAccessToken);
+            //as the accept header is needed for accessing commit search API which is still in preview mode
+            httpGet.addHeader("Accept", "application/vnd.github.cloak-preview");
+        } catch (IllegalArgumentException e) {
+            throw new CodeQualityMatricesException("The url provided for accessing the Github Search Commit API is " +
+                    "invalid ", e);
+        }
+        return ApiUtility.callApi(httpGet);
+    }
+
+    /**
+     * This is used to call the github review API.
+     *
+     * @param repoLocation      repository name
+     * @param pullRequestNumber pull request number to be queried for
+     * @param githubAccessToken Github access token for accessing github API
+     * @return String representation of the json response
+     * @throws CodeQualityMatricesException
+     */
+    public String callReviewApi(String repoLocation, int pullRequestNumber, String githubAccessToken) throws
+            CodeQualityMatricesException {
+        String url = "https://api.github.com/repos/" + repoLocation + "/pulls/" + pullRequestNumber + "/reviews";
+        try {
+            httpGet = new HttpGet(url);
+            httpGet.addHeader("Accept", "application/vnd.github.black-cat-preview+json");
+            httpGet.addHeader("Authorization", "Bearer " + githubAccessToken);
+        } catch (IllegalArgumentException e) {
+            throw new CodeQualityMatricesException("The url provided for accessing the Github Review Commit API is " +
+                    "invalid ", e);
+        }
+        return ApiUtility.callApi(httpGet);
+    }
+
+    /**
+     * This is used to call the github Issue Search API.
+     *
+     * @param commitHashToBeSearched commit hash to be searched for issues
+     * @param githubAccessToken      Github access token for accessing github API
+     * @return String representation of the json response
+     * @throws CodeQualityMatricesException
+     */
+    public String callSearchIssueApi(String commitHashToBeSearched, String githubAccessToken) throws
+            CodeQualityMatricesException {
+        String url = "https://api.github.com/search/issues?q=" + commitHashToBeSearched;
+        try {
+            httpGet = new HttpGet(url);
+            httpGet.addHeader("Accept", "application/vnd.github.mercy-preview+json");
```

Comment:
Can't we create constants for these Strings ? "Accept" and "Authorization" are used in multiple places.